packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -69,22 +69,30 @@ const decorator = (service) => ({
     const previousStage = await getEntityStage(uid, entityId);
 
     const updatedEntity = await service.update.call(this, uid, entityId, { ...opts, data });
-    if (
-      previousStage?.workflow?.id &&
-      previousStage?.id &&
-      previousStage.id !== data[ENTITY_STAGE_ATTRIBUTE]
-    ) {
-      const webhookPayload = {
-        entityId,
+
+    if (previousStage?.id && previousStage.id !== updatedEntity[ENTITY_STAGE_ATTRIBUTE].id) {
+      const model = strapi.getModel(uid);
+
+      strapi.eventHub.emit(WORKFLOW_UPDATE_STAGE, {
+        model: model.modelName,
+        uid: model.uid,
+        entity: {
+          id: entityId,
+        },
```

Comment:
minor sugestion, dont really mind but maybe simplifies the reading
```suggestion
	const updatedStage = updatedEntity[ENTITY_STAGE_ATTRIBUTE]
	
    if (previousStage?.id && previousStage.id !== updatedStage.id) {
      const model = strapi.getModel(uid);

      strapi.eventHub.emit(WORKFLOW_UPDATE_STAGE, {
        model: model.modelName,
        uid: model.uid,
        entity: {
          id: entityId,
        },
```